File: tests/src/main/java/com/github/klee0kai/test/di/earthmirror/EarthComponent.java
package com.github.klee0kai.test.di.earthmirror;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.test.mowgli.earth.Cave;

@Component(
        qualifiers = {Cave.CaveType.class, Integer.class}
)
public interface EarthComponent {

    EastModule east();

    WestModule west();

}

File: tests/src/main/java/com/github/klee0kai/test/di/earthmirror/EarthComponent.kt
package com.github.klee0kai.test.di.earthmirror

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.test.mowgli.earth.Cave.CaveType

@Component(identifiers = [CaveType::class, Int::class])
interface EarthComponent {
    fun east(): EastModule?

    fun west(): WestModule?
}
